specs/TransportProtocols.md

```diff
@@ -0,0 +1,175 @@
+# Transport Protocols
+
+This document describes the protocols used by the three ASP.NET Endpoint Transports: WebSockets, Server-Sent Events, Long Polling and HTTP Post
+
+## Transport Requirements
+
+A transport is required to have the following attributes:
+
+1. Duplex - Able to send messages from Server to Client and from Client to Server
+1. Frame-based - Messages have fixed length (as opposed to streaming, where data is just pushed throughout the connection)
+1. Frame Metadata - Able to encode two pieces of Frame Metadata
+    * `Type` - Either `Binary` or `Text`.
+    * `EndOfMessage` - Boolean indicating if this frame represents the final frame of a complete message.
```

Comment:
Ok, we'll definitely need to make EndOfMessage default to `true` so people don't accidentally forget to set it and end up never actually transmitting a frame :).

Also, WebSockets requires that all frames of a multi-frame message have the same "Opcode" (Text/Binary). We will enforce the same requirement.